src/space.cpp

```diff
@@ -809,6 +809,17 @@ static argval_t do_space(chunk_t *first, chunk_t *second, int &min_sp, bool comp
       return(cpd.settings[UO_sp_after_oc_msg_receiver].a);
    }
 
+   // c++17 structured bindings e.g., "auto [x, y, z]" vs a[x, y, z]" or "auto const [x, y, z]" vs "auto const[x, y, z]"
+   if (cpd.lang_flags & LANG_CPP
+	  && (first->type == CT_BYREF || first->type == CT_QUALIFIER || first->type == CT_TYPE)
```

Comment:
Thanks for the PR, the build fails because of the source code formatting rules, but it's not you first time so I guess you know what to do. (You will also have to change the CLI files)

Pleas take a look at 3ebb43db21c40bfbb5c4d28541daedd0897897b9, we currently don't really have that information somewhere where people could find it, or any enforcement on that but as you will have do fix the formatting anyways I would like you to also break up those `||` on new lines.